src/assets/assets.controller.ts

```diff
@@ -671,4 +675,83 @@ export class AssetsController {
 
     return handleServiceResult(result, resolver);
   }
+
+  @ApiOperation({
+    summary: 'Get Permission Groups with their permissions',
+    description: 'This endpoint allows fetching all Permission Groups with their permissions',
+  })
+  @ApiArrayResponse(PermissionGroupWithPermissionsModel, {
+    description: 'List of Permission Groups with their permissions',
+    paginated: false,
+  })
+  @ApiNotFoundResponse({
+    description: 'The Asset does not exist',
+  })
+  @Get(':asset/permission-groups')
+  public async getPermissionGroupsWithPermissions(
+    @Param() { asset }: AssetParamsDto
+  ): Promise<ResultsModel<PermissionGroupWithPermissionsModel>> {
+    const result = await this.assetsService.getPermissionGroupsWithPermissions(asset);
+
+    return new ResultsModel({
+      results: result.map(group => new PermissionGroupWithPermissionsModel(group)),
+    });
+  }
+
+  @ApiOperation({
+    summary: 'Invite external agent',
+    description: 'This endpoint invites an external agent to an Asset',
+  })
+  @ApiTransactionResponse({
+    description: 'Details about the transaction',
+    type: TransactionQueueModel,
+  })
+  @ApiNotFoundResponse({
+    description: 'The Asset does not exist',
+  })
+  @ApiTransactionFailedResponse({
+    [HttpStatus.BAD_REQUEST]: ['The target Identity is already an External Agent'],
+    [HttpStatus.UNAUTHORIZED]: [
+      'The signing identity does not have the required permissions to invite an external agent',
+    ],
+  })
+  @Post(':asset/permission-groups/invite-agent')
+  public async inviteAgent(
+    @Param() { asset }: AssetParamsDto,
+    @Body() params: InviteAgentToGroupDto
+  ): Promise<TransactionResponseModel> {
+    const result = await this.assetsService.inviteAgentToGroup(asset, params);
+
+    return handleServiceResult(result, authorizationRequestResolver);
+  }
+
+  @ApiOperation({
+    summary: 'Remove external agent',
+    description: 'This endpoint removes an external agent from an Asset',
+  })
+  @ApiTransactionResponse({
+    description: 'Details about the transaction',
+    type: TransactionQueueModel,
+  })
+  @ApiNotFoundResponse({
+    description: 'The Asset does not exist',
+  })
+  @ApiTransactionFailedResponse({
+    [HttpStatus.BAD_REQUEST]: [
+      'The target is the last Agent with full permissions for this Asset. There should always be at least one Agent with full permissions',
+      'The target Identity is not an External Agent',
+    ],
+    [HttpStatus.UNAUTHORIZED]: [
+      'The signing identity does not have the required permissions to remove an agent from a permission group',
+    ],
+  })
+  @Post(':asset/permission-groups/remove-agent')
+  public async removeAgent(
+    @Param() { asset }: AssetParamsDto,
+    @Body() params: RemoveAgentFromGroupDto
+  ): Promise<TransactionResponseModel> {
+    const result = await this.assetsService.removeAgentFromAsset(asset, params);
+
+    return handleServiceResult(result);
```

Comment:
All these APIs should be moved in its own controller - PermissionGroupController with endpoint as :asset/permission-groups